esx_service/utils/vmdk_utils.py

```diff
@@ -73,10 +73,11 @@ def init_datastoreCache(force=False):
     from local ESX host. force=True will force it to ignore current cache
     and force init
     """
+    global datastores
+
```

Comment:
I see. I was seeing a warning because of its usage in logging statement above with statement. So I moved it upwards.